parse/IntComplexValueRefParser.cpp

```diff
@@ -31,13 +72,53 @@ namespace parse {
         qi::_pass_type _pass;
         const boost::phoenix::function<detail::construct_movable> construct_movable_;
         const boost::phoenix::function<detail::deconstruct_movable> deconstruct_movable_;
-
+        const boost::phoenix::function<detail::get_pointer> get_pointer_;
+        std::string temp_s = "SHP_REINFORCED_HULL_BONUS";
+        std::string* temp_s_p = &temp_s;
+        
         game_rule
             = (   tok.GameRule_
                 > label(tok.Name_) > string_grammar
               ) [ _val = construct_movable_(new_<ValueRef::ComplexVariable<int>>(_1, nullptr, nullptr, nullptr, deconstruct_movable_(_2, _pass), nullptr)) ]
             ;
 
+        named
+            = (   tok.Named_
+                  //> label(tok.Name_) > +(qi::alnum | qi::char_('_'))
+                  > label(tok.Name_) > +(alnum)
```

Comment:
For char_ I  need to use the qi::char_type instead - not sure why this is necessary in our project (cant reproduce that behavior in a minimal parser).
 
tok.string was what I was looking for. Probably will stick to the parsed string valueref for the moment because i need it in order to construct the surrounding valueref.

Reply:
I don't think it should be necessary to have surrounding "Named" `ComplexVariable` case wrapping the actual `ValueRef`.

Reply:
Actually, it might be better to have a separate list of name-valueref pairs that is parsed specifically for the named-value lookup, rather than having them be mixed into the middle of other scripts. That way you don't need to worry about using them in macros and getting multiple-definitions of the same named valueref, and the handling of the results will be easier and less disruptive to other parsing code and scripts...

Reply:
> Actually, it might be better to have a separate list of name-valueref pairs that is parsed specifically for the named-value lookup, rather than having them be mixed into the middle of other scripts. 

Yes, I also think so but these also have downsides which should be discussed first. E.g. if we allow those to appear in any FOCS file we might have to sprinkle code in all the parsers(?). If we only allow those to appear in their own files the code gets disconnected from the use site which is probably a bad thing in 80% of the cases. Also we need to introduce a valueref which references the named valueref by name. 
I still think this is the right thing to do but i will first finish the vertical slice (i.e. named valuerefs can directly be referenced in stringtable entries and AI code)

Reply:
> I don't think it should be necessary to have surrounding "Named" `ComplexVariable` case wrapping the actual `ValueRef`.

I originally wanted to have the Named ComplexVariable in the parsed tree in order to see it, but then i saw that ComplexVariable currently does not support double valuerefs. So I said, ok i dont need it. Then I found that from the IntComplexVariableParser i have to return a ComplexVariable even if there are cases where the wrapped instance is not a Complex one - so i now have two cases there. For double it looks a bit better, could hook into DoubleVariableParser.

Then I found that all that stuff gets parsed multiple times (e.g. the reinforced hull gets parsed four times), so i ran into lifeline issues - now i keep the last registered valueref instance (which seems to work for what i tried). 
Probably not too sensible - so it seems to make it right there is no other option than the one you suggested - although i am surprised that the SHP_REINFORCED_HULL tech gets parsed four times.